From bef4add9e34a404fe16a4439678bbbf78dbc6760 Mon Sep 17 00:00:00 2001
From: Vladimir Petko <vladimir.petko@canonical.com>
Date: Fri, 27 Sep 2024 19:32:40 +1200
Subject: [PATCH] feat: gradle config

---
 build.gradle    |  8 ++++++++
 settings.gradle | 10 ++++++++++
 2 files changed, 18 insertions(+)

diff --git a/build.gradle b/build.gradle
index aa7c512..c151d64 100644
--- a/build.gradle
+++ b/build.gradle
@@ -6,6 +6,7 @@ plugins {
   id 'org.cyclonedx.bom' version '1.8.2'
   id 'io.spring.javaformat' version '0.0.41'
   id "io.spring.nohttp" version "0.0.11"
+  id "com.canonical.rockcraft" version "0.1.1"
 }
 
 apply plugin: 'java'
@@ -17,6 +18,13 @@ gradle.startParameter.excludedTaskNames += [ "checkFormatAot", "checkFormatAotTe
 group = 'org.springframework.samples'
 version = '3.3.0'
 
+rockcraft {
+    description = "readme.md"
+    summary = "Spring Petclinic Application"
+    buildPackage = 'openjdk-17-jdk-headless'
+    targetRelease = 17
+}
+
 java {
   sourceCompatibility = JavaVersion.VERSION_17
 }
diff --git a/settings.gradle b/settings.gradle
index e60ee14..949e48c 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1 +1,11 @@
+pluginManagement {
+    repositories {
+        mavenCentral {}
+        gradlePluginPortal {}
+        maven {
+            url System.getProperty('user.home') + '/.m2/repository'
+        }
+    }
+}
+
 rootProject.name = 'spring-petclinic'
-- 
2.43.0

